### PR TITLE
Terrain decorations level

### PR DESCRIPTION
**Pull request by IvanSavenko**
Sounded fun so why not? 
Change allows to specify level of terrain decorations used for terrains on RMG maps. Currently set to fixed value of 15% which seems to be similar to example screenshot from #3353. For comparison - previous level was ~66%
If somebody (e.g. Warmonger) want to tweak it or make this option configurable in editor like in H3 - go ahead, but not something that I plan to do myself.

- Implements #3353

Examples:
(note: all terrains except water & rock are affected. I use snow to show difference and because it is most noticeable here)
- Default (15%):
зображення

- No decorations (0%):
зображення

- Only decorations (100%):
зображення

### CODE REVIEW

**Comment by DjWarmonger** [comment]
Works great for me :)

File: lib/rmg/threadpool/MapProxy.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ class MapProxy
     void insertObjects(std::set<CGObjectInstance*>& objects);
     void removeObject(CGObjectInstance* obj);
 
-    void drawTerrain(CRandomGenerator & generator, std::vector<int3> & tiles, TerrainId terrain);
+	void drawTerrain(CRandomGenerator & generator, std::vector<int3> & tiles, TerrainId terrain);
```
Comment (DjWarmonger) [inline lib/rmg/threadpool/MapProxy.h:31]: Unneccessary tab

**Review by DjWarmonger — APPROVED** [review]

**Comment by IvanSavenko** [comment]
Cool. Will merge to 1.4.2 then

> Also, works for new terrains without modifications.

... as long as new terrains use same tiles layout

**Comment by misiokles** [comment]
Highlands and Wastelands have more tiles as far I remember.

**Comment by IvanSavenko** [comment]
@misiokles 
Should work I think?
hota terrains have custom tileset in vcmi - specifically because they have more tiles.
I did adjusted ranges for decorations/plain tiles for hota tileset, so should work (although have not checked in game)

File: lib/rmg/threadpool/MapProxy.h (hunk 1 above)
Comment (IvanSavenko) [reply lib/rmg/threadpool/MapProxy.h:31]: Actually looks like entire file uses spaces instead of tabs (except for 1 line changed by me)

**Pull request merged**